src/services/locationService.js

```diff
@@ -39,4 +39,88 @@ async function createLocation(campus, building_id, floor_number, location_type_i
 
 }
 
-export default {createLocation}
\ No newline at end of file
+async function deleteLocation(location_id) {
+    const conn = await database.connect();
+    
+    try {
+
+        const sql = "UPDATE locations SET soft_delete = 1 WHERE location_id = ?";
+        await conn.query(sql, [location_id]);
+        return{ success: true, message: `Localização deletada com sucesso.`};
+    } catch (error) {
+        console.error("Erro ao deletar localização:", error);
+        return {success: false, message:'Erro ao remover a localização'};
+    } finally {
+        conn.end();
+    }
+}
+
+async function updateLocation(location_id, campus, building_id, floor_number, location_type_id, location_name, description){
+    const conn = await database.connect();
+
+    try {
+        
+        const [existingBuilding] = await conn.query('SELECT 1 FROM buildings WHERE building_id = ? AND soft_delete = 0', [building_id]);
+
+        if(existingBuilding.length === 0){
+            return { success: false, message: 'O prédio especificado não existe.' };
+        }
+
+        const [existingLocationTypeId] = await conn.query('SELECT 1 FROM location_types WHERE type_id = ? AND soft_delete = 0', [location_type_id]);
+
+        if(existingLocationTypeId.length === 0){
+            return { success: false, message: 'O tipo de localização não existe.' };
+        }
+
+        const [existingLocations] = await conn.query('SELECT location_id FROM locations WHERE location_name = ? AND building_id = ? AND soft_delete = 0', [location_name, building_id]);
+
+        if(existingLocations.length > 0) {
+            return { success: false, message: 'Já existe uma localização com esse nome neste prédio.', data: existingLocations };
+        }
+
+        const locationData = 'UPDATE locations SET campus = ?, building_id = ?, floor_number = ?, location_type_id = ?, location_name = ?, description = ? WHERE location_id = ? AND soft_delete = 0';
+
+        const dataLocation = [campus, building_id, floor_number, location_type_id, location_name, description, location_id];
+
+        await conn.query(locationData, dataLocation);
+
+        return { success: true, message: `Localização ${location_name} atualizada com sucesso.`}
+
+    } catch (error) {
+        return { success: false, message: error.message}
+    } finally{
+        conn.end()
+    }
+}
+
+async function getLocation(location_id){
+    const conn = await database.connect();
+
+    try{
+    
+        const sql = "SELECT * FROM locations WHERE location_id = ? AND soft_delete = 0";
+        const [rows] = await conn.query(sql, [location_id]);
+        
+        if(rows.length > 0){
+            return { success: true, data: rows };
+        } else{
+            return { success: false, message: 'Nenhuma localização encontrada com esse ID.' }
+        }
+        
+    } catch(error){
+        return { success: false, message: error.message}
+    } finally{
+        conn.end();
+    }
```

Comment:
Lê primeiro o comentário na controller!

Percebe que aqui você retorna outro sucesso? Eu entendi o propósito mas simplesmente não é necessário nenhum desses sucessos pois isso é trabalho do código HTTP. O que isso ta fazendo é apenas criar um json maior e mais díficil de ser acessado, o json pode ficar assim se for retornado apenas rows:

![image](https://github.com/user-attachments/assets/cb045caf-a3d4-41ec-9d14-a78a87587dfb)
O sucesso http (200) já faz todo o trabalho da variável sucess mandada tlg

Reply:
E os meus eu fiz retornar apenas a data, sem a estrutura "message" primeiro, vamos falar hj na etec como padronizar isso, pois é um dos problemas que moiou na integração com o front